### PR TITLE
Reintroduces Python3.7 support

### LINKED ISSUE #122: Reintroduce support for Python 3.7
Hi! A quick question:

Why have you dropped `python3.7` support in this commit https://github.com/antonagestam/phantom-types/commit/c0817edd886237d288b277c93a89c04d286e54e1 ?

Looks like `python3.7` was supported not a long time ago: https://github.com/antonagestam/phantom-types/commit/c2e6d37a0afa2a9dd531df016414a12688341006

From what I can see the only feature of `python3.8` you are using is `/` arguments, which can be easily converted into regular arguments (which is even cleaner in my opinion). Anything else that I've missed?

### PR DESCRIPTION
**Pull request by sobolevn**
Closes #122

### CODE REVIEW

**Comment by sobolevn** [comment]
Btw, checkout https://pypi.org/project/typing-utils/#issubtype instead of https://github.com/antonagestam/phantom-types/blob/main/phantom/utils.py#L70

**Comment by sobolevn** [comment]
There's a metaclass conflict on `3.7` here: https://github.com/antonagestam/phantom-types/blob/main/phantom/sized.py#L61-L68

```
______________ ERROR collecting tests/pydantic/test_pydantic_schemas.py ______________
tests/pydantic/test_pydantic_schemas.py:16: in <module>
    from phantom.sized import Empty
phantom/sized.py:67: in <module>
    abstract=True,
E   TypeError: metaclass conflict: the metaclass of a derived class must be a (non-strict) subclass of the metaclasses of all its bases
```

Any ideas? There are so much going on there, that I am a bit lost 🙂

**Comment by antonagestam** [comment]
@sobolevn Not sure, but perhaps importing from `typing_extensions` would work? https://github.com/antonagestam/phantom-types/commit/c2e6d37a0afa2a9dd531df016414a12688341006#diff-9bdde33f4b2d15f7803bd0dd947fbf07a5bc65bb253ca6ffb53bb40b44f99cd0R12-R19

The `_ProtocolMeta` situation is a bit messy.

**Comment by sobolevn** [comment]
It is still failing with some strange messages:

```
======================================= ERRORS =======================================
______________ ERROR collecting tests/pydantic/test_pydantic_schemas.py ______________
tests/pydantic/test_pydantic_schemas.py:44: in <module>
    class DataModel(pydantic.BaseModel):
pydantic/main.py:299: in pydantic.main.ModelMetaclass.__new__
    ???
pydantic/fields.py:411: in pydantic.fields.ModelField.infer
    ???
pydantic/fields.py:342: in pydantic.fields.ModelField.__init__
    ???
pydantic/fields.py:451: in pydantic.fields.ModelField.prepare
    ???
pydantic/fields.py:520: in pydantic.fields.ModelField._type_analysis
    ???
phantom/base.py:44: in __instancecheck__
    return self.__instancecheck__(instance)
phantom/base.py:201: in __instancecheck__
    if cls.__abstract__:
E   AttributeError: type object 'Empty' has no attribute '__abstract__'
______________ ERROR collecting tests/pydantic/test_pydantic_schemas.py ______________
tests/pydantic/test_pydantic_schemas.py:44: in <module>
    class DataModel(pydantic.BaseModel):
pydantic/main.py:299: in pydantic.main.ModelMetaclass.__new__
    ???
pydantic/fields.py:411: in pydantic.fields.ModelField.infer
    ???
pydantic/fields.py:342: in pydantic.fields.ModelField.__init__
    ???
pydantic/fields.py:451: in pydantic.fields.ModelField.prepare
    ???
pydantic/fields.py:520: in pydantic.fields.ModelField._type_analysis
    ???
phantom/base.py:44: in __instancecheck__
    return self.__instancecheck__(instance)
phantom/base.py:201: in __instancecheck__
    if cls.__abstract__:
E   AttributeError: type object 'Empty' has no attribute '__abstract__'
============================== short test summary info ===============================
ERROR tests/pydantic/test_pydantic_schemas.py - AttributeError: type object 'Empty'...
ERROR tests/pydantic/test_pydantic_schemas.py - AttributeError: type object 'Empty'...
!!!!!!!!!!!!!!!!!!!!!! Interrupted: 2 errors during collection !!!!!!!!!!!!!!!!!!!!!!!
================================= 2 errors in 1.02s ==================================
make: *** [coverage] Error 2
```

File: setup.cfg [hunk 1]
```diff
@@ -21,9 +23,10 @@ url = https://github.com/antonagestam/phantom-types/
 [options]
 include_package_data = True
 packages = find:
-python_requires = >=3.8
+python_requires = >=3.7
 install_requires =
     typeguard>=2.10
+    typing_extensions>=3.10
```
Comment (sobolevn) [inline setup.cfg:29]: This was already a part of `typeguard`, but I wanted to make this explicit.

**Comment by sobolevn** [comment]
Ok, I've tried several hacks and all of them didn't work to solve the metaclass conflict. Any ideas?

**Comment by antonagestam** [comment]
@sobolevn I don't have any ideas off the top of my head, I'd have to go back to the old code and look at that.

File: docs/requirements.in [hunk 2]
```diff
@@ -4,6 +4,7 @@ recommonmark
 m2r2
 mistune<2
 typeguard
-furo
+furo==2021.4.11b34
```
Comment (sobolevn) [inline docs/requirements.in:7]: This was failing with 

```
» make requirements
Could not find a version that matches sphinx<3.2,>=1.3.1,>=3.0,>=3.1,~=4.0 (from -r requirements.in (line 1))
Tried: 0.1.61611, 0.1.61798, 0.1.61843, 0.1.61945, 0.1.61950, 0.2, 0.3, 0.4, 0.4.1, 0.4.2, 0.4.3, 0.5, 0.5.1, 0.5.2, 0.6, 0.6.1, 0.6.2, 0.6.3, 0.6.4, 0.6.5, 0.6.6, 0.6.7, 1.0, 1.0.1, 1.0.2, 1.0.3, 1.0.4, 1.0.5, 1.0.6, 1.0.7, 1.0.8, 1.1, 1.1.1, 1.1.2, 1.1.3, 1.2, 1.2, 1.2, 1.2.1, 1.2.1, 1.2.1, 1.2.2, 1.2.2, 1.2.2, 1.2.3, 1.2.3, 1.3, 1.3, 1.3.1, 1.3.1, 1.3.2, 1.3.2, 1.3.3, 1.3.3, 1.3.4, 1.3.4, 1.3.5, 1.3.5, 1.3.6, 1.3.6, 1.4, 1.4, 1.4.1, 1.4.1, 1.4.2, 1.4.2, 1.4.3, 1.4.3, 1.4.4, 1.4.4, 1.4.5, 1.4.5, 1.4.6, 1.4.6, 1.4.7, 1.4.7, 1.4.8, 1.4.8, 1.4.9, 1.4.9, 1.5, 1.5, 1.5.1, 1.5.1, 1.5.2, 1.5.2, 1.5.3, 1.5.3, 1.5.4, 1.5.4, 1.5.5, 1.5.5, 1.5.6, 1.5.6, 1.6.1, 1.6.1, 1.6.2, 1.6.2, 1.6.3, 1.6.3, 1.6.4, 1.6.4, 1.6.5, 1.6.5, 1.6.6, 1.6.6, 1.6.7, 1.6.7, 1.7.0, 1.7.0, 1.7.1, 1.7.1, 1.7.2, 1.7.2, 1.7.3, 1.7.3, 1.7.4, 1.7.4, 1.7.5, 1.7.5, 1.7.6, 1.7.6, 1.7.7, 1.7.7, 1.7.8, 1.7.8, 1.7.9, 1.7.9, 1.8.0, 1.8.0, 1.8.1, 1.8.1, 1.8.2, 1.8.2, 1.8.3, 1.8.3, 1.8.4, 1.8.4, 1.8.5, 1.8.5, 2.0.0, 2.0.0, 2.0.1, 2.0.1, 2.1.0, 2.1.0, 2.1.1, 2.1.1, 2.1.2, 2.1.2, 2.2.0, 2.2.0, 2.2.1, 2.2.1, 2.2.2, 2.2.2, 2.3.0, 2.3.0, 2.3.1, 2.3.1, 2.4.0, 2.4.0, 2.4.1, 2.4.1, 2.4.2, 2.4.2, 2.4.3, 2.4.3, 2.4.4, 2.4.4, 3.0.0, 3.0.0, 3.0.1, 3.0.1, 3.0.2, 3.0.2, 3.0.3, 3.0.3, 3.0.4, 3.0.4, 3.1.0, 3.1.0, 3.1.1, 3.1.1, 3.1.2, 3.1.2, 3.2.0, 3.2.0, 3.2.1, 3.2.1, 3.3.0, 3.3.0, 3.3.1, 3.3.1, 3.4.0, 3.4.0, 3.4.1, 3.4.1, 3.4.2, 3.4.2, 3.4.3, 3.4.3, 3.5.0, 3.5.0, 3.5.1, 3.5.1, 3.5.2, 3.5.2, 3.5.3, 3.5.3, 3.5.4, 3.5.4, 4.0.0, 4.0.0, 4.0.1, 4.0.1, 4.0.2, 4.0.2, 4.0.3, 4.0.3, 4.1.0, 4.1.0
Skipped pre-versions: 0.5.2b1, 0.6b1, 1.0b1, 1.0b2, 1.2b1, 1.2b2, 1.2b3, 1.3b1, 1.3b1, 1.3b2, 1.3b2, 1.3b3, 1.3b3, 1.4a1, 1.4a1, 1.4b1, 1.4b1, 1.5a1, 1.5a1, 1.5a2, 1.5a2, 1.5b1, 1.5b1, 1.6b1, 1.6b1, 1.6b2, 1.6b2, 1.6b3, 1.6b3, 1.7.0b1, 1.7.0b1, 1.7.0b2, 1.7.0b2, 1.8.0b1, 1.8.0b1, 2.0.0b1, 2.0.0b1, 2.0.0b2, 2.0.0b2, 3.0.0b1, 3.0.0b1, 4.0.0b1, 4.0.0b1, 4.0.0b2, 4.0.0b2
There are incompatible versions in the resolved dependencies:
  sphinx<3.2,>=3.1 (from -r requirements.in (line 1))
  sphinx>=1.3.1 (from recommonmark==0.7.1->-r requirements.in (line 3))
  Sphinx>=3.0 (from sphinx-autodoc-typehints==1.12.0->-r requirements.in (line 2))
  sphinx~=4.0 (from furo==2021.7.5b38->-r requirements.in (line 7))
make: *** [requirements] Error 2
```

**Review by antonagestam — COMMENTED** [review]
Nice work! 👍 I only have one issue in that I want to keep ´[mypy].python_version´ in setup.cfg.

File: setup.cfg [hunk 3]
```diff
@@ -55,7 +58,6 @@ src_paths = phantom, tests
 force_single_line = True
 
 [mypy]
-python_version = 3.8
```
[the remark this answers is not in the document]
Comment (sobolevn) [reply setup.cfg:58]: Can you please explain why (I am still not sure which solution is best)? 🙂

**Comment by sobolevn** [comment]
Thanks a lot for your help! 👍

**Comment by antonagestam** [comment]
@sobolevn Thanks for putting in the work, the help is very appreciated ⭐️

**Comment by antonagestam** [comment]
I'll publish this in 0.10.0.

**Pull request merged**